injector/grpc_disruption.go

```diff
@@ -0,0 +1,105 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021 Datadog, Inc.
+
+package injector
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/DataDog/chaos-controller/api/v1beta1"
+	chaos_grpc "github.com/DataDog/chaos-controller/grpc"
+	pb "github.com/DataDog/chaos-controller/grpc/disruption_listener"
+	"google.golang.org/grpc"
+)
+
+// GRPCDisruptionInjector describes a grpc disruption
+type GRPCDisruptionInjector struct {
+	spec       v1beta1.GRPCDisruptionSpec
+	config     GRPCDisruptionInjectorConfig
+	serverAddr string
+}
+
+// GRPCDisruptionInjectorConfig contains all needed drivers to create a grpc disruption
+type GRPCDisruptionInjectorConfig struct {
+	Config
+}
+
+// NewGRPCDisruptionInjector creates a GRPCDisruptionInjector object with the given config,
+// missing fields are initialized with the defaults
+func NewGRPCDisruptionInjector(spec v1beta1.GRPCDisruptionSpec, config GRPCDisruptionInjectorConfig) (Injector, error) {
+	return GRPCDisruptionInjector{
+		spec:       spec,
+		config:     config,
+		serverAddr: config.TargetPodIP + ":" + strconv.Itoa(spec.Port),
+	}, nil
+}
+
+// Inject injects the given dns disruption into the given container
+func (i GRPCDisruptionInjector) Inject() error {
+	i.config.Log.Infow(fmt.Sprintf("%t", i.config.DryRun))
+
+	i.config.Log.Infow("connecting to " + i.serverAddr + "...")
+
+	if i.config.DryRun {
+		i.config.Log.Infow("adding dry run mode grpc disruption", "spec", i.spec)
+		return nil
+	}
+
+	conn, err := connectToServer(i.serverAddr)
+	if err != nil {
+		i.config.Log.Fatalf(err.Error())
```

Comment:
I am pretty sure that a fatal log kills the process, and thus we wouldnt end up returning this error afterwards in the next line? Should this just be an error log?